src/app/(pages)/queryBuilding/querySelection/QueryLibrary.tsx

```diff
@@ -32,41 +30,43 @@ import {
 
 interface UserQueriesDisplayProps {
   queries: CustomUserQuery[];
-  selectedQuery: SelectedQueryState;
-  setSelectedQuery: Dispatch<SetStateAction<SelectedQueryDetails>>;
   setBuildStep: Dispatch<SetStateAction<BuildStep>>;
 }
 
 /**
  * Component for query building when user-generated queries already exist
  * @param root0 - The props object.
  * @param root0.queries - Array of user-generated queries to display.
- * @param root0.selectedQuery - the query object we're building
  * @param root0.setBuildStep - setter function to progress the stage of the query
  * building flow
- * @param root0.setSelectedQuery - setter function to update the query for editing
  * @returns the UserQueriesDisplay to render the queries with edit/delete options
  */
 export const MyQueriesDisplay: React.FC<UserQueriesDisplayProps> = ({
   queries: initialQueries,
-  selectedQuery,
-  setSelectedQuery,
   setBuildStep,
 }) => {
-  const queriesContext = useContext(DataContext);
+  const queryContext = useContext(DataContext);
+  if (!queryContext?.setSelectedQuery) {
```

Comment:
instead of throwing here in the body of the component, could we try / catch in the `handleEdit` function and using the toast pattern we have elsewhere? Think in cases where a dev messed up, the only thing that would be broken would be the update which we'd indicate to users via a toast flow.

This feels like a more graceful fallback compared to rendering the general "client side error" page